nc2zarr/processor.py

```diff
@@ -37,39 +39,52 @@ def process_dataset(self, ds: xr.Dataset) -> Tuple[xr.Dataset, Dict[str, Dict[st
         if self._process_rename:
             ds = ds.rename(self._process_rename)
         if self._process_rechunk:
-            chunk_encoding = self._get_chunk_encodings(ds, self._process_rechunk)
+            ds, chunk_encoding = self._rechunk_dataset(ds, self._process_rechunk)
         else:
             chunk_encoding = dict()
         return ds, self._merge_encodings(ds,
                                          chunk_encoding,
                                          self._output_encoding or {})
 
     @classmethod
-    def _get_chunk_encodings(cls,
-                             ds: xr.Dataset,
-                             process_rechunk: Dict[str, Dict[str, int]]) \
-            -> Dict[str, Dict[str, Any]]:
+    def _rechunk_dataset(cls,
+                         ds: xr.Dataset,
+                         process_rechunk: Dict[str, Dict[str, int]]) \
+            -> Tuple[xr.Dataset, Dict[str, Dict[str, Any]]]:
+        ds_rechunked = ds.copy()
         output_encoding = dict()
-        all_chunk_sizes = process_rechunk.get('*', {})
+        all_dim_chunk_sizes = process_rechunk.get('*', {})
         for k, v in ds.variables.items():
             var_name = str(k)
-            var_chunk_sizes = dict(all_chunk_sizes)
-            var_chunk_sizes_delta = process_rechunk.get(var_name, {})
-            if var_chunk_sizes_delta is None:
-                var_chunk_sizes_delta = {dim_name: None for dim_name in v.dims}
-            elif isinstance(var_chunk_sizes_delta, int):
-                var_chunk_sizes_delta = {var_name: var_chunk_sizes_delta}
-            var_chunk_sizes.update(var_chunk_sizes_delta)
+            # Compute default chunk sizes for dims of v
+            dim_chunk_sizes = dict(all_dim_chunk_sizes)
+            dim_chunk_sizes_update = process_rechunk.get(var_name, _UNDEFINED)
+            if dim_chunk_sizes_update is not _UNDEFINED:
+                if dim_chunk_sizes_update is None \
+                        or isinstance(dim_chunk_sizes_update, int) \
+                        or dim_chunk_sizes_update == 'input':
+                    dim_chunk_sizes_update = {dim_name: dim_chunk_sizes_update for dim_name in v.dims}
+                elif isinstance(dim_chunk_sizes_update, dict):
```

Comment:
If you do change the Dict types to Mappings, this should be updated to

```suggestion
                elif isinstance(dim_chunk_sizes_update, Mapping):
```